db/migrate/20150803065716_create_materials.rb

```diff
@@ -0,0 +1,24 @@
+class CreateMaterials < ActiveRecord::Migration
+  def change
+    create_table :course_material_folders do |t|
+      t.references :parent_folder, foreign_key: { references: :course_material_folders }
+      t.references :owner, polymorphic: true
+      t.string :name, null: false
+      t.text :description
+      t.datetime :start_at, null: false
```

Comment:
I think every folder should have a start time ?  And if you want the folder to always be available, just set the start time earlier than current time.